### ISSUE TITLE
via WP forum: Notice: Undefined index: option_page

### ISSUE THREAD

**Issue by senlin** [senlin/so-clean-up-wp-seo#37]
originally posted by [antonv](https://wordpress.org/support/users/antonv/) on [WP.org support](https://wordpress.org/support/topic/notice-undefined-index-option_page/):

WordPress 4.9.1, PHP 7.1, Hide SEO Bloat 3.4.1
and in wp-config.php: define(‘WP_DEBUG’, true);

If I edit a User and then Save I get following warnings:

Notice: Undefined index: option_page in /home/btelg/public_html/wp-content/plugins/so-clean-up-wp-seo/includes/class-so-clean-up-wp-seo-settings.php on line 102

Warning: Cannot modify header information – headers already sent by (output started at /home/btelg/public_html/wp-content/plugins/so-clean-up-wp-seo/includes/class-so-clean-up-wp-seo-settings.php:102) in /home/btelg/public_html/wp-admin/includes/misc.php on line 1114

Warning: Cannot modify header information – headers already sent by (output started at /home/btelg/public_html/wp-content/plugins/so-clean-up-wp-seo/includes/class-so-clean-up-wp-seo-settings.php:102) in /home/btelg/public_html/wp-includes/pluggable.php on line 1216

**Comment by senlin** [comment]
@afragen 
Do you have any idea what this can be about? Perhaps to do with PHP 7.1? Or is it the `option_page` reference on line 102 of `includes/class-so-clean-up-wp-seo-settings.php`?

Hope you have an idea...

**Comment by afragen** [comment]
I’ll take a look.

**Comment by afragen** [comment]
Adding something like the following above that conditional should work. You could even put it at the beginning of the function as an early exit.

```php
if ( ! isset( $_POST[‘option_page’], $_POST[‘action’] ) ) {
    return;
}
```

Should fix the issue. 

I typed this on an iPad so you may need to replace single quotes if they are stylized and spacing might be incorrect.

**Comment by senlin** [comment]
Thanks Andy!

I made a separate branch to ask the person who opened the issue to test it, as I cannot reproduce it.

So far I have not heard anything back from that person, if I don't hear anything should I merge [the new branch](https://github.com/senlin/so-clean-up-wp-seo/tree/issue37) or not?

Also unrelated to this, wish you a very Happy New Year!
Cheers,
Piet

**Comment by afragen** [comment]
Give the OP some time to respond. I find I'm always in more of a rush to fix the issue than some with the issue are in a rush to test the fix. 😉 

If you don't hear anything by the end of next week I'd merge. It's clearly not effecting everyone and the patch is safe.

Also, you don't need the `GitHub Branch` header any more. GHU will automatically save the branch on Install from GHU or on branch switch.

**Comment by senlin** [comment]
Thanks Andy, good advice and a good tip! :)

**Comment by anton-vrba** [comment]
Hi Guys, Happy New Year

Well done! Fix37 did fix it. (I was editing a user, manually changed role and password)

It was not a PHP issue as I switched to 5.6 and had the same Notice and Warning.

**Comment by senlin** [comment]
@anton-vrba Happy New Year Anton!

Thanks for testing and confirming it fixed the issue.

We have just merged the branch and released 3.5.0 of the plugin here and on [wp.org](https://wordpress.org/plugins/so-clean-up-wp-seo/)
